tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/telemetryMetrics/util/InMemoryMetricReader.java

```diff
@@ -0,0 +1,116 @@
+    boolean isShutdown = false;
+
+    public static InMemoryMetricReader current() {
+        return CDI.current().select(InMemoryMetricReader.class).get();
+    }
+
+    @Override
+    public AggregationTemporality getAggregationTemporality(InstrumentType instrumentType) {
+        return AggregationTemporality.CUMULATIVE;
+    }
+
+    @Override
+    public void register(CollectionRegistration registration) {
+        if (isShutdown) {
+            throw new IllegalStateException("InMemoryMetricReader has been shutdown");
+        }
+
+        collectionRegistration = registration;
+    }
+
+    @Override
+    public CompletableResultCode forceFlush() {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Unimplemented method 'forceFlush'");
+    }
+
+    @Override
+    public CompletableResultCode shutdown() {
+        collectionRegistration = null;
+        isShutdown = true;
+        return CompletableResultCode.ofSuccess();
+    }
+
+    public long readLongData(MetricID id) {
+        @SuppressWarnings("unchecked")
+        List<LongPointData> longData = (List<LongPointData>) getPointData(id);
+
+        return longData.stream()
+                .mapToLong(LongPointData::getValue)
+                .sum();
+    }
+
+    protected List<?> getPointData(MetricID id) {
+        Collection<MetricData> allMetrics = collectionRegistration.collectAllMetrics();
+
+        return allMetrics.stream()
+                .filter(
+                        md -> md.getName().equals(id.name))
+                .flatMap(
+                        md -> md.getData().getPoints().stream())
+                .filter(
+                        point -> id.attributes.asMap().keySet().stream()
+                                .allMatch(key -> point.getAttributes().asMap().containsKey(key)
+                                        && id.attributes.asMap().get(key)
+                                                .equals(point.getAttributes().asMap().get(key))))
+                .collect(Collectors.toList());
+    }
+
+    public Optional<LongPointData> getGaugueMetricLatestValue(MetricID id) {
+        Collection<MetricData> allMetrics = collectionRegistration.collectAllMetrics();
+
+        Optional<LongPointData> gague = allMetrics.stream()
+                .filter(
+                        md -> md.getName().equals(id.name))
+                .flatMap(md -> md.getLongGaugeData().getPoints().stream())
+                .filter(point -> id.attributes.asMap().keySet().stream()
+                        .allMatch(key -> point.getAttributes().asMap().containsKey(key)
+                                && id.attributes.asMap().get(key)
+                                        .equals(point.getAttributes().asMap().get(key))))
+                .sorted((pointOne, pointTwo) -> Long.compare(pointOne.getEpochNanos(), pointTwo.getEpochNanos()))
+                .findFirst();
+
+        return gague;
+
+    }
```

Comment:
```java
                .sorted((pointOne, pointTwo) -> Long.compare(pointOne.getEpochNanos(), pointTwo.getEpochNanos()))
                .findFirst();
```

Does this not sort the points in ascending order of time and pick the first one, which would give you the earliest value rather than the latest value?